src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -132,21 +132,40 @@ public int OnActiveProjectCfgChange(IVsHierarchy pIVsHierarchy)
         #endregion
 
         #region IDisposable
-        public void Dispose()
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
         {
-            // Build manager APIs require UI thread access.
-            _threadingService.ExecuteSynchronously(async() =>
+            if (!disposedValue)
             {
-                await _threadingService.SwitchToUIThread();
-
-                if (_buildManager != null)
+                if (disposing)
                 {
-                    // Unregister solution build events.
-                    _buildManager.UnadviseUpdateSolutionEvents(_solutionEventsCookie);
-                    _buildManager = null;
+                    // Build manager APIs require UI thread access.
+                    _threadingService.ExecuteSynchronously(async () =>
+                    {
+                        await _threadingService.SwitchToUIThread();
+
+                        if (_buildManager != null)
+                        {
+                            // Unregister solution build events.
+                            _buildManager.UnadviseUpdateSolutionEvents(_solutionEventsCookie);
+                            _buildManager = null;
+                        }
+                    });
                 }
-            });
+
+                disposedValue = true;
+            }
         }
+
+        // This code added to correctly implement the disposable pattern.
```

Comment:
Avoid the comments - it's obvious you're implement the dispose pattern.

Reply:
Why is this rule running on internal types?

Reply:
technically any type that can be inherited from should implement this pattern.  Just because we're the only ones that will ever extend this type doesn't mean that we'll always get it right.